graphql_compiler/global_utils.py

```diff
@@ -29,3 +31,16 @@ def is_same_type(left, right):
         return is_same_type(left.of_type, right.of_type)
     else:
         return False
+
+
+def generate_new_name(base_name: str, taken_names: Sequence[str]) -> str:
+    """Return a name based on the provided string that is not already taken.
+
+    This method tries the following names:
+    {base_name}_0, then {base_name}_1, etc.
+    and returns the first one that's not in taken_names
```

Comment:
Reminder about our rule that docstrings that span more than one line always have `Args` and `Returns` sections.

Reply:
I thought we got rid of that rule since mypy. In this case it just adds noise (6 redundant lines).